aten/src/ATen/core/op_registration/op_registration.h

```diff
@@ -18,10 +18,23 @@ namespace c10 {
 
 namespace detail {
 template<class KernelFunctor>
-std::unique_ptr<FunctionSchema> inferFunctionSchemaFromFunctor() {
+std::unique_ptr<FunctionSchema> inferFunctionSchemaFromFunctor(
+          std::enable_if_t<!c10::guts::infer_function_traits_t<KernelFunctor>::first_arg_is_dispatchKeySet::value, std::nullptr_t> = nullptr) {
   using func_type = typename c10::guts::infer_function_traits_t<KernelFunctor>::func_type;
   return std::make_unique<FunctionSchema>(inferFunctionSchemaFlattenedReturns<func_type>("", ""));
 }
+
+// Specialization for when the first argument is of type DispatchKeySet
+// We do this because every argument in a function schema is expected to be convertable
+// to an ivalue, but DispatchKeySet is not a type we want the jit to be aware of.
+// See Note [Plumbing Keys Through The Dispatcher]
+template<class KernelFunctor>
+std::unique_ptr<FunctionSchema> inferFunctionSchemaFromFunctor(
```

Comment:
we wouldn't need this specialization if we called inferFunctionSchemaFromFunctor on the internal representation of the function pointer, i.e. the one that always has the dispatch key argument. Not sure which approach is better though.

Reply:
Hmm... that is true. Although that won't solve the problem for `CppSignature::make()`, because we explicitly want to hide the DispatchKeySet from jit. So taking this suggestion on its own won't be enough for me to get rid of `function_traits::func_type_skip_first_arg` (which is the reason I added a specialization for `function_traits`).